streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1202,16 +1202,47 @@ public class StreamsConfig extends AbstractConfig {
         PRODUCER_EOS_OVERRIDES = Collections.unmodifiableMap(tempProducerDefaultOverrides);
     }
 
+    private static final Map<String, Object> KS_DEFAULT_CONSUMER_CONFIGS;
+    static {
+        final Map<String, Object> tempConsumerDefaultOverrides = new HashMap<>();
+        tempConsumerDefaultOverrides.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, "1000");
+        tempConsumerDefaultOverrides.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        tempConsumerDefaultOverrides.put("internal.leave.group.on.close", false);
+
+        KS_CONTROLLED_CONSUMER_CONFIGS = Collections.unmodifiableMap(tempConsumerDefaultOverrides);
```

Comment:
We do allow users to modify these configs. I don't think we want to lock it down?

Reply:
`KS_DEFAULT_CONSUMER_CONFIGS` Is a map of defaults that Kafka Streams prefers. It does not lock these properties. The user can overwrite these configs. 

`KS_CONTROLLED_CONSUMER_CONFIGS` and `KS_CONTROLLED_CONSUMER_CONFIGS_EOS_ENABLED` are the configs we do not allow being overwritten. KS will overwrite the user configs in the end